pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -354,54 +375,93 @@ func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLis
 	taskRun.Status.PodName = pod.Name
 
 	taskRun.Status.Steps = []v1alpha1.StepState{}
+	// Might be complete if we're only running because of a sidecar
+	maybeComplete := len(pod.Status.ContainerStatuses) > 0 && pod.Status.Phase == corev1.PodRunning
```

Comment:
hm im a bit confused about what "maybe complete" means here - would calling this something like `stepsComplete` make sense?

Reply:
I've refactored this part of the function a little bit to make it slightly more readable.